### PR TITLE
Adding image tasks evaluation examples

### PR DESCRIPTION
**Pull request by shubhamiit**
# Description
Adding image classification evluation examples

Runs:
1. MC: https://ml.azure.com/experiments/id/7d94b7c2-b985-4056-9ca5-998722e752da/runs/loyal_ghost_9z9xj3l5c3?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rupaljain-rs/providers/Microsoft.MachineLearningServices/workspaces/rupaljain-workspace&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#
2. ML: https://ml.azure.com/experiments/id/0374dac4-9da9-4d66-92c0-b34355c3abb5/runs/icy_hamster_m00wqlpdsj?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rupaljain-rs/providers/Microsoft.MachineLearningServices/workspaces/rupaljain-workspace&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#
3. OD: https://ml.azure.com/experiments/id/737e3b93-ff19-4dea-aff9-9e5043a8adc6/runs/green_napa_dyvdrlv25w?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rupaljain-rs/providers/Microsoft.MachineLearningServices/workspaces/rupaljain-workspace&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#
4. IS: https://ml.azure.com/experiments/id/446c5fae-3d4c-4cd0-91cf-b6147026411e/runs/purple_hominy_6ptldwk0rl?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rupaljain-rs/providers/Microsoft.MachineLearningServices/workspaces/rupaljain-workspace&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [x] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/python/foundation-models/system/evaluation/image-multiclass-classification/image-multiclass-classification.ipynb [hunk 1]
```diff
@@ -0,0 +1,694 @@
...
+   "source": [
+    "## Image Multi-class Classification Evaluation\n",
+    "\n",
+    "This sample shows how use the evaluate a group of models against a given set of metrics for the `image-classification` task. \n",
+    "\n",
+    "### Evaluation dataset\n",
+    "We will use the [fridgeObjects](https://cvbp-secondary.z19.web.core.windows.net/datasets/image_classification/fridgeObjects.zip) dataset.\n",
+    "\n",
+    "### Model\n",
+    "The goal of evaluating models is to compare their performance on a variety of metrics. `image-classification` is generic task type. As such, the models you pick to compare must be finetuned for same scenario. Given that we have the dataset, we would like to look for models finetuned for this specific scenario. We will compare `microsoft-beit-base-patch16-224-pt22k-ft22k` and `microsoft-swinv2-base-patch4-window12-192-22k` in this sample, which are available in the `azureml-preview` system registry.\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/evaluation/image-multiclass-classification/image-multiclass-classification.ipynb:16]:
```suggestion
    "The goal of evaluating models is to compare their performance on a variety of metrics. `image-classification` is a generic task type. As such, the models you pick to compare must be finetuned for the same scenario. Given that we have the dataset, we would like to look for models finetuned for this specific scenario. We will compare `microsoft-beit-base-patch16-224-pt22k-ft22k` and `microsoft-swinv2-base-patch4-window12-192-22k` in this sample, which are available in the `azureml-preview` system registry.\n",
```

File: sdk/python/foundation-models/system/evaluation/image-multiclass-classification/image-multiclass-classification.ipynb [hunk 2]
```diff
@@ -0,0 +1,694 @@
...
+    "\n",
+    "If you'd like to evaluate models that are not in the system registry, you can import those models to your workspace or organization registry and then evaluate them using the approach outlined in this sample. Review the sample notebook for [importing models](https://github.com/Azure/azureml-examples/blob/main/sdk/python/foundation-models/system/import/import_model_into_registry.ipynb). \n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/evaluation/image-multiclass-classification/image-multiclass-classification.ipynb:18]: Use relative path instead

File: sdk/python/foundation-models/system/evaluation/image-multiclass-classification/image-multiclass-classification.ipynb [hunk 3]
```diff
@@ -0,0 +1,694 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install azure-ai-ml\n",
+    "%pip install azure-identity\n",
+    "%pip install datasets==2.9.0"
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/evaluation/image-multiclass-classification/image-multiclass-classification.ipynb:47]: Do we need datasets?

File: sdk/python/foundation-models/system/evaluation/image-multiclass-classification/image-multiclass-classification.ipynb [hunk 4]
```diff
@@ -0,0 +1,694 @@
...
+    "\n",
+    "try:\n",
+    "    model_evaluation_compute = workspace_ml_client.compute.get(model_evaluation_cluster_name)\n",
+    "    print(\"Found existing compute target.\")\n",
+    "except ResourceNotFoundError:\n",
+    "    print(\"Creating a new compute target...\")\n",
+    "    model_evaluation_compute = AmlCompute(\n",
+    "        name=model_evaluation_cluster_name,\n",
+    "        type=\"amlcompute\",\n",
+    "        size=\"Standard_NC6\",\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/evaluation/image-multiclass-classification/image-multiclass-classification.ipynb:145]: We want to use Standard_NC6s_v3 instead

File: sdk/python/foundation-models/system/evaluation/image-multiclass-classification/image-multiclass-classification.ipynb [hunk 5]
```diff
@@ -0,0 +1,694 @@
...
+    "\n",
+    "metrics_df = pd.DataFrame()\n",
+    "\n",
+    "for job in pipeline_jobs:\n",
+    "    # Concat 'tags.mlflow.rootRunId=' and pipeline_job.name in single quotes as filter variable\n",
+    "    filter = \"tags.mlflow.rootRunId='\" + job[\"job_name\"] + \"'\"\n",
+    "    runs = mlflow.search_runs(experiment_names=[experiment_name], filter_string=filter, output_format=\"list\")\n",
+    "\n",
+    "    # Get the training and evaluation runs.\n",
+    "    for run in runs:\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/evaluation/image-multiclass-classification/image-multiclass-classification.ipynb:660]: Is this snippet working now?

File: sdk/python/foundation-models/system/evaluation/image-object-detection/fridge-eval-config.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
+  "metrics": ["mean_average_precision", "recall", "precision", "per_label_metrics", "image_level_binary_classsifier_metrics", "confusion_matrices_per_score_threshold"],
+  "class_labels": ["can", "carton", "milk_bottle", "water_bottle"],
```
Comment (MadhuM02) [inline sdk/python/foundation-models/system/evaluation/image-object-detection/fridge-eval-config.json:3]: I think we are only reading metrics, iou_threshold and box_score_threshold from this config

**Comment by MadhuM02** [comment]
can you please link all the runs??

File: sdk/python/foundation-models/system/evaluation/image-classification/multiclass-classification/README.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
+## Multiclass Classification
+
+### List of supported keyword arguments:
+
+|     Keyword Argument     | Description                                                                    | Type             | Sample                                                          |
+|:------------------------:|:-------------------------------------------------------------------------------|------------------|-----------------------------------------------------------------|
+|         metrics          | List for subset of metrics to be computed. All supported metrics listed below. | list<str>        | ["accuracy", "f1_score_micro", "average_precision_score_macro"] |
+|       class_labels       | List for superset of all existing labels in our dataset                        | list, np.ndarray | [0, 1, 2, 3], ["cat", "dog", "panda"]        |
```
Comment (gauravrajguru) [inline sdk/python/foundation-models/system/evaluation/image-classification/multiclass-classification/README.md:8]: we can show parameter is optional or not?

File: sdk/python/foundation-models/system/evaluation/image-classification/multiclass-classification/image-multiclass-classification.ipynb [hunk 8]
```diff
@@ -0,0 +1,687 @@
...
+   "source": [
+    "import os\n",
+    "\n",
+    "# We'll copy each JSONL file within its related MLTable folder\n",
+    "validation_mltable_path = os.path.join(dataset_parent_dir, \"validation-mltable-folder\")\n",
+    "\n",
+    "# First, let's create the folders if they don't exist\n",
+    "os.makedirs(validation_mltable_path, exist_ok=True)\n",
+    "\n",
+    "train_validation_ratio = 5\n",
```
Comment (gauravrajguru) [inline sdk/python/foundation-models/system/evaluation/image-classification/multiclass-classification/image-multiclass-classification.ipynb:409]: nit: whole dataset can be used for evaluation.

File: sdk/python/foundation-models/system/evaluation/image-object-detection/fridge-eval-config.json [hunk 9]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (gauravrajguru) [inline sdk/python/foundation-models/system/evaluation/image-object-detection/fridge-eval-config.json:1]: is eval-config mandatory (for my knowledge)?

Why iou_thresh and box_score are not part of this config.

**Review by gauravrajguru — APPROVED** [review]
minor comments.

**Review by MadhuM02 — APPROVED** [review]

**Comment by rjaincc** [comment]
Please remove me from reviewers, @shubhamiit. I've made changes in the PR.

**Pull request merged**